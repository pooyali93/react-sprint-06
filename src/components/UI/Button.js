import React from "react";
import PropTypes  from "prop-types";
import './Button.scss'

export default function Button({ iconName, color, text,  onClick, }) {

    return (
      <button  style={{backgroundColor:color}} className="btn" onClick={onClick}>
        {iconName} {text}
      </button>
    );
  };

  Button.defaultProps = {
    color:'red',

}

Button.prototype = {
    text:PropTypes.string,
    color:PropTypes.string,
    onclick:PropTypes.func,
}