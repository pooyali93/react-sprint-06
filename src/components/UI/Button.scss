
$primary-color: rgb(40, 40, 66);
$secondary-color:rgb(58, 110, 165);
$gray-color:rgb(192, 192, 192);
$light-color:rgb(235, 235, 235);
$orange-color:rgb(209, 69, 50);
$black-color:rgb(22, 26, 29);


.btn {
    display: inline-block;
    background-color:var($primary-color);
    color: var($light-color);
    border: none;
    padding: 10px 40px;
    margin: 5px;
    border-radius: 5px;
    cursor: pointer;
    text-decoration: none;
    font-size: 15px;
    font-family: inherit;
  

  .btn:focus {
    outline: none;
  }
  
  .btn:active {
    transform: scale(0.98);
  }
  
  .btn-block {
    display: block;
    width: 100%;
  }
}


