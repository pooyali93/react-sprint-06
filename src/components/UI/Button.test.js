import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'; // add this line

import Button from './Button';

describe('Matches snapshot', () => {
  const tree = renderer.create(<Button value="submit" />).toJSON();
  expect(tree).toMatchSnapshot();
});
