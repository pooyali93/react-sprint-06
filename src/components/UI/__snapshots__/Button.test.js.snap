// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="button"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
   
</button>
`;
